src/common/schedulers/slurm_commands.py

```diff
@@ -97,6 +97,25 @@ def update_nodes(
     if we run scontrol update state=fail_state nodeaddr=good_addr nodename=name,
     the scontrol command will fail but nodeaddr will be updated to good_addr.
     """
+    # Only split on , if there is ] before
+    # For ex. "node-[1,3,4-5],node-[20,30]" should split into ["node-[1,3,4-5]","node-[20,30]"]
+    nodes = re.split("(?<=]),", nodes) if type(nodes) is str else nodes
+    nodeaddrs = re.split("(?<=]),", nodeaddrs) if type(nodeaddrs) is str else nodeaddrs
+    nodehostnames = re.split("(?<=]),", nodehostnames) if type(nodehostnames) is str else nodehostnames
```

Comment:
probably you don't need a regex split for nodeaddrs and nodehostnames but simply need to split on ","

Reply:
I removed the code. My code is only executed if if nodeaddrs or hostnames are are/is being changed. The input in this case should be list already